blocks/library/image/index.js

```diff
@@ -72,8 +72,20 @@ export const settings = {
 				isMatch( node ) {
 					const tag = node.nodeName.toLowerCase();
 					const hasImage = node.querySelector( 'img' );
-
-					return tag === 'img' || ( hasImage && tag === 'figure' );
+					const isAnchorWithOneImage = ( nodeToCheck ) =>
+						'A' === nodeToCheck.nodeName &&
+						1 === nodeToCheck.childNodes.length &&
+						'IMG' === nodeToCheck.firstChild.nodeName;
+
+					const isParagraphWithAnchorImage =
+						'P' ===	node.nodeName &&
+						1 === node.childNodes.length &&
+						isAnchorWithOneImage( node.firstChild );
+
+					return tag === 'img' ||
+						( hasImage && tag === 'figure' ) ||
+						isParagraphWithAnchorImage ||
```

Comment:
Hm we already removed this check for normal images by unwrapping beforehand... Feel too bad to have to do it again for anchors. I almost wonder if it wouldn't be easier to add a data attribute or something during image normalisation, which might be a lot easier and clearer.

Reply:
Thank you for the review @iseulde. I had another working approach where I normalized <a><img ...></a> to <figure><a><img ...></a></figure>. And that one worked without changing image block. But I thought not adding a new normalizer would be more consensual, I will add commit so we can decide on the best option :)